src/object_gui.cpp

```diff
@@ -25,23 +29,35 @@
 
 #include "safeguards.h"
 
-static ObjectClassID _selected_object_class; ///< the currently visible object class
-static int _selected_object_index;           ///< the index of the selected object in the current class or -1
-static uint8 _selected_object_view;          ///< the view of the selected object
+static const ObjectClass *_selected_object_class;  ///< Currently selected available object class.
+static int _selected_object_index;                 ///< Index of the currently selected object if existing, else \c -1.
+static uint8 _selected_object_view;                ///< the view of the selected object
 
 /** The window used for building objects. */
 class BuildObjectWindow : public Window {
-	static const int OBJECT_MARGIN = 4; ///< The margin (in pixels) around an object.
-	int line_height;                    ///< The height of a single line.
-	int info_height;                    ///< The height of the info box.
-	Scrollbar *vscroll;                 ///< The scrollbar.
+	typedef GUIList<const ObjectClass*, StringFilter&> GUIObjectClassList;
+
+	static const uint EDITBOX_MAX_SIZE = 16;
```

Comment:
Missing documentation. 16 pixels or characters? :)

Reply:
Fixed, resolving. Specified it's characters :)